definitions/3.0.0/messageObject.json

```diff
@@ -154,6 +154,90 @@
       }
     }
   },
+  "allOf": [
+    {
+      "if": {
+        "not": {
+          "required": [
+            "schemaFormat"
+          ]
+        }
+      },
+      "then": {
+        "properties": {
+          "payload": {
+            "$ref": "http://asyncapi.com/definitions/3.0.0/schema.json"
+          }
+        }
+      }
+    },
+    {
+      "if": {
+        "required": [
+          "schemaFormat"
+        ],
+        "properties": {
+          "schemaFormat": {
+            "enum": [
+              "application/vnd.aai.asyncapi;version=2.0.0",
+              "application/vnd.aai.asyncapi+json;version=2.0.0",
+              "application/vnd.aai.asyncapi+yaml;version=2.0.0",
+              "application/vnd.aai.asyncapi;version=2.1.0",
+              "application/vnd.aai.asyncapi+json;version=2.1.0",
+              "application/vnd.aai.asyncapi+yaml;version=2.1.0",
+              "application/vnd.aai.asyncapi;version=2.2.0",
+              "application/vnd.aai.asyncapi+json;version=2.2.0",
+              "application/vnd.aai.asyncapi+yaml;version=2.2.0",
+              "application/vnd.aai.asyncapi;version=2.3.0",
+              "application/vnd.aai.asyncapi+json;version=2.3.0",
+              "application/vnd.aai.asyncapi+yaml;version=2.3.0",
+              "application/vnd.aai.asyncapi;version=2.4.0",
+              "application/vnd.aai.asyncapi+json;version=2.4.0",
+              "application/vnd.aai.asyncapi+yaml;version=2.4.0",
+              "application/vnd.aai.asyncapi;version=2.5.0",
+              "application/vnd.aai.asyncapi+json;version=2.5.0",
+              "application/vnd.aai.asyncapi+yaml;version=2.5.0",
+              "application/vnd.aai.asyncapi;version=2.6.0",
+              "application/vnd.aai.asyncapi+json;version=2.6.0",
+              "application/vnd.aai.asyncapi+yaml;version=2.6.0",
```

Comment:
Because there are no changes to the schema across all v2 versions and v3, I guess it makes sense to validate it against these, right?

Reply:
I'd say it does make sense. Better to be explicit ;]

Reply:
I'm concerned about having to maintain this list, add new elements per version, etc.

Can we use a pattern instead of enum here?

```json
{
  "type": "string",
  "pattern": "^application\\/vnd\\.aai\\.asyncapi(\\+json|\\+yaml)?;version=\\d+\\.\\d+\\.\\d+$"
}
```

Maybe some change is needed. You can test it in https://www.jsonschemavalidator.net/s/VTaBCoGr

Reply:
Pretty sure we dont want to enable v1 or future v4 schemas here right? 🤔

Reply:
Also would we not want to explicitly state which schema formats are allowed as values? Using something like `v2.69.0` would not give a validation error with regex 🤔

Unless you explicitly add a range, but then you are back to having to change this each version anyway.

Reply:
Good point.

Also thinking about new features being added between versions (no breaking changes). For example, let's say `3.1.0` introduces a new field in the schema so you will have to do a split, serving one schema for >=2.x <=3.0, and another for 3.1.

Porting this to the pattern will be a bit messy. I don't like to maintain all this list but indeed i can't find a better alternative ATM but automating somehow this (we can think about in near future).